### PR TITLE
Add Unit Tests to model_service_worker

### PR DESCRIPTION
**Pull request by zachgk**
Issue 759

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: ci/Dockerfile.python2.7 [hunk 1]
```diff
@@ -179,6 +179,7 @@ RUN set -ex \
     && pip install retrying \
     && pip install mock \
     && pip install pytest -U \
+    && pip install pytest-mock -U \
```
Comment (piyushghai) [inline ci/Dockerfile.python2.7]: These shouldn't be there. The pytest-mock install should go in buildspec.yml

File: mms/tests/unit_tests/test_model_service_worker.py [hunk 2]
```diff
@@ -0,0 +1,606 @@
...
+
+            sock.close.assert_called()
+            patches.log.assert_called_with(log_call_param)
+
+        def test_stop_server(self, patches, worker, sock):
+            worker.stop_server(sock)
+            worker.send_response.assert_called()
+            sock.close.assert_called()
+
+
```
Comment (piyushghai) [inline mms/tests/unit_tests/test_model_service_worker.py]: Too many extra lines here.

File: mms/tests/unit_tests/test_model_service_worker.py [hunk 3]
```diff
@@ -0,0 +1,606 @@
...
+
+
+def test_emit_metrics(mocker):
+    metrics = {'test_emit_metrics': True}
+    printer = mocker.patch('mms.model_service_worker.print')
+    dumps = mocker.patch('json.dumps')
+    flush = mocker.patch('sys.stdout')
```
Comment (piyushghai) [inline mms/tests/unit_tests/test_model_service_worker.py]: What are we doing with the flush mocker ?

File: mms/model_service_worker.py [hunk 4]
```diff
@@ -91,10 +92,10 @@ def create_predict_response(self, ret, req_id_map, invalid_reqs):
                 result.update({"requestId": req_id_map[idx]})
                 result.update({"code": 200})
 
-                if isinstance(val, bytes):
-                    value = ModelWorkerCodecHelper.encode_msg(encoding, val)
-                elif isinstance(val, str):
+                if isinstance(val, str):
```
[the remark this answers is not in the document]
Comment (zachgk) [reply mms/model_service_worker.py]: In python2, str and bytes are considered the same.  We switched them while investigating this and attempting to achieve full coverage.  Ultimately, python2 will register the second option uncovered during testing regardless of the order, but it does at least return the correct result.

**Pull request merged**